src/App.js

```diff
@@ -1,43 +1,99 @@
-import Header from './components/ui/Header';
 import ListadoActividades from './components/actividades/ListadoActividades';
 import ListadoEspacios from './components/espacios/ListadoEspacios';
+import Login from './components/login/Login';
+import Registro from './components/registro/Registro';
+import PropTypes from 'prop-types';
 
 export default function App() {
+  const [estaAutorizado, setEstaAutorizado] = useState(false);
+
   return (
     <Container>
       <Header />
+      {console.log(estaAutorizado)}
       <Box my={4}>
         <Router>
           <Switch>
-            <Route exact path="/">
-              <ListadoEspacios />
+            <Route exact path="/" />
+
+            <Route path="/login">
+              <Login setEstaAutorizado={setEstaAutorizado} />
             </Route>
-            <Route exact path="/espacios">
-              <ListadoEspacios />
+
+            <Route path="/registro">
+              <Registro setEstaAutorizado={setEstaAutorizado} />
             </Route>
-            <Route path="/espacios/nuevo">
+
+            <PrivateRoute
+              exact
+              path="/espacios"
+              isAuthenticated={estaAutorizado}
+            >
+              <ListadoEspacios />
+            </PrivateRoute>
+
+            <PrivateRoute
+              path="/espacios/nuevo"
+              isAuthenticated={estaAutorizado}
+            >
               <AltaModificacionEspacio titulo={'Carga de espacios'} />
-            </Route>
-            <Route path="/espacios/:id">
+            </PrivateRoute>
+
+            <PrivateRoute path="/espacios/:id" isAuthenticated={estaAutorizado}>
               <AltaModificacionEspacio titulo={'Modificar espacio'} />
-            </Route>
-            <Route exact path="/actividades">
+            </PrivateRoute>
+
+            <PrivateRoute
+              exact
+              path="/actividades"
+              isAuthenticated={estaAutorizado}
+            >
               <ListadoActividades />
-            </Route>
-            <Route path="/actividades/nueva">
+            </PrivateRoute>
+
+            <PrivateRoute
+              path="/actividades/nueva"
+              isAuthenticated={estaAutorizado}
+            >
               <AltaModificacionActividad titulo={'Carga de actividades'} />
-            </Route>
-            <Route path="/actividades/:id">
+            </PrivateRoute>
+
+            <PrivateRoute
+              path="/actividades/:id"
+              isAuthenticated={estaAutorizado}
+            >
               <AltaModificacionActividad titulo={'Modificar actividad'} />
-            </Route>
+            </PrivateRoute>
           </Switch>
         </Router>
       </Box>
     </Container>
   );
 }
+
+function PrivateRoute({ children, isAuthenticated, ...rest }) {
+  return (
+    <Route
+      {...rest}
+      render={() => {
+        return isAuthenticated === true ? children : <Redirect to={'/login'} />;
+      }}
```

Comment:
Un detalle pavo, se puede escribir sin llaves:

```suggestion
      render={() => isAuthenticated ? children : (<Redirect to={'/login'} />)}
```

Reply:
Ah, y el `=== true` es error de intro che. 😛

Reply:
Te juro que el === true lo sacamos... habrá sido uno de esos Control + Z. No se quiso ir